Cant load Microsoft.Identity.Client.dll on an Ubuntu Azure DevOps Agent

### Verified issue does not already exist?

Yes

### What error did you receive?

````
 153 |  … Invoke($script:PSModuleRoot, "$(Join-Path $script:DllRoot smo)", $scr …
     |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "Invoke" with "3" argument(s): "The running
     | command stopped because the preference variable
     | "ErrorActionPreference" or common parameter is set to Stop:
     | Could not import
     | /root/.local/share/powershell/Modules/dbatools/1.1.7/bin/smo/coreclr/Microsoft.Identity.Client.dll :  MethodInvocationException: /root/.local/share/powershell/Modules/dbatools/1.1.7/internal/scripts/libraryimport.ps1:143 Line |  143 |                  [Reflection.Assembly]::LoadFrom($assemblyPath)      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~      
| Exception calling "LoadFrom" with "1" argument(s): "Assembly      | with same name is already loaded"  "
````
### Steps to Reproduce

Created a brand new Azure DevOps Pipeline in my own org using ubuntu-latest 
Created a step using Azure PowerShell task

````
pool:
  vmImage: ubuntu-latest

steps:
- task: AzurePowerShell@5
  inputs:
    azureSubscription: 'BeardSubscription'
    ScriptType: 'InlineScript'
    Inline: |
      Install-Module dbatools -Scope CurrentUser -Force
      Import-Module dbatools -Verbose
    azurePowerShellVersion: 'LatestVersion'
 ````


### Are you running the latest release?

Yes

### Other details or mentions



The resolution for now was to pin the Install and Import to Version 1.0.173

### What PowerShell host was used when producing this error

PowerShell Core (pwsh.exe)

### PowerShell Host Version

It will be PowerShell 7

### SQL Server Edition and Build number

N/A

Just created a dummy pipeline in my org which I have access to. Here is the full log

````
Starting: AzurePowerShell
==============================================================================
Task         : Azure PowerShell
Description  : Run a PowerShell script within an Azure environment
Version      : 5.185.0
Author       : Microsoft Corporation
Help         : https://aka.ms/azurepowershelltroubleshooting
==============================================================================
Generating script.
/usr/bin/pwsh -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command . '/home/vsts/work/_temp/945a4e8b-bc39-4f3e-8bf7-1c8d5aebe6ea.ps1'
Saved!
Import-Module -Name /usr/share/az_6.1.0/Az.Accounts/2.5.2/Az.Accounts.psd1 -Global
Clear-AzContext -Scope Process
Clear-AzContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue
Connect-AzAccount -ServicePrincipal -Tenant *** -Credential System.Management.Automation.PSCredential -Environment AzureCloud @processScope
 Set-AzContext -SubscriptionId  -TenantId ***
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Installing package 'dbatools'                                                      Downloaded 0.00 MB out of 29.09 MB.                                             [                                                                    ]                                                                                                                                                                       Installing package 'dbatools'                                                      Downloaded 2.92 MB out of 29.09 MB.                                             [oooo                                                                ]                                                                                                                                                                       Installing package 'dbatools'                                                      Downloaded 5.84 MB out of 29.09 MB.                                             [ooooooooo                                                           ]                                                                                                                                                                       Installing package 'dbatools'                                                      Downloaded 14.61 MB out of 29.09 MB.                                            [ooooooooooooooooooooooo                                             ]                                                                                                                                                                       Installing package 'dbatools'                                                      Downloaded 20.45 MB out of 29.09 MB.                                            [ooooooooooooooooooooooooooooooooo                                   ]                                                                                                                                                                       Installing package 'dbatools'                                                      Unzipping                                                                       [ooooooooooooooooooooooooooooooooooooooooooooooo                     ]                                                                                                                                                                       Installing package 'dbatools'                                                      Copying unzipped package to '/tmp/447846962/dbatools.1.1.8'                     [ooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo       ]                                                                                                                                                                       Installing package 'dbatools'                                                      Process Package Manifest                                                        [oooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo    ]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      ParentContainsErrorRecordException: /home/vsts/.local/share/powershell/Modules/dbatools/1.1.8/internal/scripts/libraryimport.ps1:153
Line |
 153 |  … Invoke($script:PSModuleRoot, "$(Join-Path $script:DllRoot smo)", $scr …
     |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "Invoke" with "3" argument(s): "The running
     | command stopped because the preference variable
     | "ErrorActionPreference" or common parameter is set to Stop:
     | Could not import
     | /home/vsts/.local/share/powershell/Modules/dbatools/1.1.8/bin/smo/coreclr/Microsoft.Identity.Client.dll :  MethodInvocationException: /home/vsts/.local/share/powershell/Modules/dbatools/1.1.8/internal/scripts/libraryimport.ps1:143 Line |  143 |                  [Reflection.Assembly]::LoadFrom($assemblyPath)      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~      | Exception calling "LoadFrom" with "1" argument(s): "Assembly      | with same name is already loaded"  "

##[error]PowerShell exited with code '1'.
Finishing: AzurePowerShell
````

So it starts by importing the Az.Accounts module which I guess is where that dll is loaded. Which means that @shueybubbles idea of altering the order of loading wont work.

--- a/internal/scripts/libraryimport.ps1
+++ b/internal/scripts/libraryimport.ps1
@@ -104,6 +104,7 @@ $scriptBlock = {
     $shared = @()
     $separator = [IO.Path]::DirectorySeparatorChar
     $shared += "third-party" + $separator + "Bogus" + $separator + "Bogus"
+    $assemblies = [System.AppDomain]::CurrentDomain.GetAssemblies()
 
     foreach ($name in $shared) {
         $assemblyPath = "$script:PSModuleRoot" + $separator + "bin\libraries" + $separator + "$name.dll"
@@ -111,9 +112,6 @@ $scriptBlock = {
         $null = try {
             Import-Module $assemblyPath
         } catch {
-            #Write-Warning $assemblyPath
-            #Write-Warning "$psitem"
-            #Write-Warning "~~~~~~~~~~~~~~~~~~~`n`n"
             try {
                 [Reflection.Assembly]::LoadFrom($assemblyPath)
             } catch {
@@ -133,16 +131,16 @@ $scriptBlock = {
         }
         Copy-Assembly -ModuleRoot $ModuleRoot -DllRoot $DllRoot -DoCopy $DoCopy -Name $name
         $assemblyPath = "$basepath$([IO.Path]::DirectorySeparatorChar)$name.dll"
-        $null = try {
-            Import-Module $assemblyPath
-        } catch {
-            #Write-Warning $assemblyPath
-            #Write-Warning "$psitem"
-            #Write-Warning "~~~~~~~~~~~~~~~~~~~`n`n"
-            try {
-                [Reflection.Assembly]::LoadFrom($assemblyPath)
+
+        if ($assemblies.FullName -notmatch "$name,") {
+            $null = try {
+                Import-Module $assemblyPath
             } catch {
-                Write-Error "Could not import $assemblyPath : $($_ | Out-String)"
+                try {
+                    [Reflection.Assembly]::LoadFrom($assemblyPath)
+                } catch {
+                    Write-Error "Could not import $assemblyPath : $($_ | Out-String)"
+                }
             }
         }
     }

--- a/internal/scripts/libraryimport.ps1
+++ b/internal/scripts/libraryimport.ps1
@@ -132,7 +132,7 @@ $scriptBlock = {
         Copy-Assembly -ModuleRoot $ModuleRoot -DllRoot $DllRoot -DoCopy $DoCopy -Name $name
         $assemblyPath = "$basepath$([IO.Path]::DirectorySeparatorChar)$name.dll"
 
-        if ($assemblies.FullName -notmatch "$name,".Replace("win\", "")) {
+        if (-not (($assemblies.FullName | Out-String).Contains("$name,".Replace("win\", "")))) {
             $null = try {
                 Import-Module $assemblyPath
             } catch {

--- a/internal/scripts/libraryimport.ps1
+++ b/internal/scripts/libraryimport.ps1
@@ -132,7 +132,7 @@ $scriptBlock = {
         Copy-Assembly -ModuleRoot $ModuleRoot -DllRoot $DllRoot -DoCopy $DoCopy -Name $name
         $assemblyPath = "$basepath$([IO.Path]::DirectorySeparatorChar)$name.dll"
 
-        if ($assemblies.FullName -notmatch "$name,") {
+        if ($assemblies.FullName -notmatch "$name,".Replace("win\", "")) {
             $null = try {
                 Import-Module $assemblyPath
             } catch {

